src/main/java/seedu/jarvis/logic/parser/planner/AddTaskCommandParser.java

```diff
@@ -35,11 +35,11 @@ public AddTaskCommand parse(String args) throws ParseException {
                                                                     PREFIX_FREQ, PREFIX_PRIORITY, PREFIX_TAG);
 
         if (!arePrefixesPresent(argMultimap, PREFIX_TASK_TYPE, PREFIX_TASK_DES)
-            || !argMultimap.getPreamble().isEmpty() || (isEventOrDeadline(argMultimap)
-                && !arePrefixesPresent(argMultimap, PREFIX_DATE))) {
-
+            || !argMultimap.getPreamble().isEmpty()
+            || (isEventOrDeadline(argMultimap) && !arePrefixesPresent(argMultimap, PREFIX_DATE))) {
```

Comment:
It would be clearer to abstract `isEventOrDeadline(argMultimap) && !arePrefixesPresent(argMultimap, PREFIX_DATE))` into a `isValidEventOrDeadline` check, as it is confusing as to why `isEventOrDeadline` is being checked twice here and later on. Initially it looked like a redundant check, so readability of logic here is quite low.

From what I can see, you first guard condition is to evaluate whether the essential prefixes are present for tasks, and in addition, if they are tasks or deadlines, they need a date prefix. I think it can be explained clearer with better abstractions